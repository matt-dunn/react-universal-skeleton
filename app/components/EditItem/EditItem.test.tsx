import React from "react";
import {shallow} from "enzyme";

import nextState from "components/state-mutate-with-status";

import {EditItem as Component} from "./index";

describe("EditItem", () => {
    it("renders correctly", () => {
        const item = {
            id: "item-1",
            name: "Item 1"
        };

        const wrapper = shallow(
            <Component
                item={item}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it("sets type 'primary'", () => {
        const item = {
            id: "item-1",
            name: "Item 1"
        };

        const wrapper = shallow(
            <Component
                item={item}
                type="primary"
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it("sets type 'secondary'", () => {
        const item = {
            id: "item-1",
            name: "Item 1"
        };

        const wrapper = shallow(
            <Component
                item={item}
                type="secondary"
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it("Sets processing message", () => {
        const item = nextState({
            id: "item-1",
            name: "Item 1"
        }, {
            type: "xx",
            meta: {
                $status: {
                    transactionId: "1",
                    processing: true,
                    complete: false
                }
            }
        });

        const wrapper = shallow(
            <Component
                item={item}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it("Sets error message", () => {
        const item = nextState({
            id: "item-1",
            name: "Item 1"
        }, {
            type: "xx",
            meta: {
                $status: {
                    transactionId: "1",
                    processing: false,
                    complete: false,
                    error: {
                        message: "Mock error"
                    }
                }
            }
        });

        const wrapper = shallow(
            <Component
                item={item}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
