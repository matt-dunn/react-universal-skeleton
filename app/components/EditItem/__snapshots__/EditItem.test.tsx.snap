// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditItem Sets error message 1`] = `
<div>
  <Container>
    <Value
      indicateInvalid={true}
      label="Item"
      multiline={true}
      onChange={[Function]}
      value="Item 1"
    />
  </Container>
  <Message>
    Mock error
  </Message>
</div>
`;

exports[`EditItem Sets processing message 1`] = `
<div>
  <Container>
    <Value
      indicateInvalid={true}
      label="Item"
      multiline={true}
      onChange={[Function]}
      value="Item 1"
    />
  </Container>
  <Message>
    Saving...
  </Message>
</div>
`;

exports[`EditItem renders correctly 1`] = `
<div>
  <Container>
    <Value
      indicateInvalid={true}
      label="Item"
      multiline={true}
      onChange={[Function]}
      value="Item 1"
    />
  </Container>
</div>
`;

exports[`EditItem sets type 'primary' 1`] = `
<div>
  <Container
    type="primary"
  >
    <Value
      indicateInvalid={true}
      label="Item"
      multiline={true}
      onChange={[Function]}
      value="Item 1"
    />
  </Container>
</div>
`;

exports[`EditItem sets type 'secondary' 1`] = `
<div>
  <Container
    type="secondary"
  >
    <Value
      indicateInvalid={true}
      label="Item"
      multiline={true}
      onChange={[Function]}
      value="Item 1"
    />
  </Container>
</div>
`;
